### PR TITLE
clean up 0.8 whatsnew, other minor doc issues

### LINKED ISSUE #1009: change whats new API section names
I put an API change in the wrong section of the whats new file in #1008. In case it's a more common misreading, I thought I should check how other packages label these sections. 

pvlib:
* API Changes with Deprecations
* API Changes

Pandas:
* Backwards incompatible API changes
* Deprecations

Numpy:
* Expired deprecations
* Compatibility notes
* Deprecations
* C API changes
* Changes

xarray:
* Breaking changes
* Deprecations

For pvlib, I think xarray's pattern is most appropriate. I don't like the similarity of pvlib's sections.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #1009 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Please let me know if I've missed any contributors to issue/PR discussions that were closed in this release.

### CODE REVIEW

File: docs/sphinx/source/api.rst [hunk 1]
```diff
@@ -379,7 +378,8 @@ Effects on PV System Output
 ===========================
 
 Loss models
-^^^^^^^^^^^
+-----------
```
Comment (kanderso-nrel) [inline docs/sphinx/source/api.rst:381]: `Loss models` is the only heading under `Effects on PV System Output`.  Should the functions be brought up a level?

**Review by kanderso-nrel — COMMENTED** [review]
A handful of nits

File: docs/sphinx/source/api.rst (hunk 1 above)
Comment (wholmgren) [reply docs/sphinx/source/api.rst:383]: I added more subheadings below.

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 2]
```diff
@@ -1,58 +1,42 @@
...
   :py:func:`pvlib.iotools.read_tmy3`, :py:meth:`pvlib.location.Location.from_tmy`, and
   :py:class:`pvlib.pvsystem.LocalizedPVSystem` for alternatives. (:issue:`965`)
   (:pull:`1008`)
+
 * The following functions, methods, and arguments were deprecated in a previous
   release and have now been removed (:issue:`966`, :pull:`1033`):
+
     * ``pvsystem.PVSystem.ashraeiam``. Use :py:meth:`pvlib.pvsystem.PVSystem.get_iam`.
     * ``pvsystem.PVSystem.physicaliam``. Use :py:meth:`pvlib.pvsystem.PVSystem.get_iam`.
     * ``pvsystem.PVSystem.sapm_aoi_loss``. Use :py:meth:`pvlib.pvsystem.PVSystem.get_iam`.
```
[the remark this answers is not in the document]
Comment (wholmgren) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:42]: I added entries to `api.rst` for the methods referenced in this PR.

We're missing a lot of pages for `PVSystem` methods. See lack of links in the methods section here: https://pvlib-python.readthedocs.io/en/latest/generated/pvlib.pvsystem.PVSystem.html

If I recall correctly, I struggled to get sphinx autosummary/autodoc to make those pages a couple of years ago.

[the remark this answers is not in the document]
Comment (wholmgren) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:42]: https://github.com/pvlib/pvlib-python/issues/1055

**Comment by wholmgren** [comment]
@cwhanse @mikofski @kanderso-nrel any final edits? I plan to tag 0.8.0 once this is merged.

Rendered whats new: https://pvlib-python--1054.org.readthedocs.build/en/1054/whatsnew.html

**Comment by mikofski** [comment]
This bullet in enhancements could be formatted:
>Added *racking_model*, *module_type*, and *temperature_model_parameters* to PVSystem, LocalizedPVSystem, SingleAxisTracker, and LocalizedSingleAxisTracker repr methods. (GH1027)

I think the arguments should have double backticks and the classes should be linked to module indeed

**Review by mikofski — APPROVED** [review]
LGTM! Thanks Will!

**Comment by cwhanse** [comment]
Only edit I saw, @mikofski noted.

**Comment by wholmgren** [comment]
I should have listened to @kanderso-nrel when he caught that in the PR review. If that's a rule it should be in the Contributing/Documentation section.

**Review by kanderso-nrel — APPROVED** [review]

**Pull request merged**